crates/runtime/src/datafusion/refresh_sql.rs

```diff
@@ -55,49 +86,377 @@ pub fn validate_refresh_sql(expected_table: TableReference, refresh_sql: &str) -
+                                        expected_table,
+                                    }
+                                );
+                            }
+                            GroupByExpr::Expressions(exprs, modifiers) => {
+                                ensure!(
+                                    exprs.is_empty(),
+                                    UnexpectedExpressionSnafu {
+                                        expr: "GROUP BY",
+                                        expected_table,
+                                    }
+                                );
+                                ensure!(
+                                    modifiers.is_empty(),
+                                    UnexpectedExpressionSnafu {
+                                        expr: "GROUP BY",
+                                        expected_table,
+                                    }
+                                );
+                            }
                         }
-                        _ => {
-                            InvalidSqlStatementSnafu { expected_table }.fail()?;
+                        ensure!(
+                            select.having.is_none(),
+                            UnexpectedExpressionSnafu {
+                                expr: "HAVING",
+                                expected_table,
+                            }
+                        );
+                        ensure!(
+                            select.into.is_none(),
+                            UnexpectedExpressionSnafu {
+                                expr: "INTO",
+                                expected_table,
+                            }
+                        );
+                        ensure!(
+                            select.lateral_views.is_empty(),
+                            UnexpectedExpressionSnafu {
+                                expr: "LATERAL VIEW",
+                                expected_table,
+                            }
+                        );
+                        ensure!(
+                            select.named_window.is_empty(),
+                            UnexpectedExpressionSnafu {
+                                expr: "WINDOW",
+                                expected_table,
+                            }
+                        );
+                        ensure!(
+                            select.prewhere.is_none(),
+                            UnexpectedExpressionSnafu {
+                                expr: "PREWHERE",
+                                expected_table,
+                            }
+                        );
+                        ensure!(
+                            select.qualify.is_none(),
+                            UnexpectedExpressionSnafu {
+                                expr: "QUALIFY",
+                                expected_table,
+                            }
+                        );
```

Comment:
could we make this a `macro_rules!` pattern? it's repeated quite a lot in similar ways

Reply:
Addressed